src/Roave/InfectionStaticAnalysis/Bootstrapper.php

```diff
@@ -18,18 +18,17 @@ public static function bootstrap(
     ): Container {
         $reflectionOffsetSet = (new ReflectionMethod(Container::class, 'offsetSet'));
 
-        // @TODO we may want to make this one lazy, but for now this is OK
-        $replacedService = $container->getMutantExecutionResultFactory();
-        $factory         = static function () use ($replacedService, $runStaticAnalysis): MutantExecutionResultFactory {
+        $new_container = clone $container;
+        $factory       = static function () use ($container, $runStaticAnalysis): MutantExecutionResultFactory {
             return new RunStaticAnalysisAgainstEscapedMutant(
-                $replacedService,
-                $runStaticAnalysis
+                $container->getMutantExecutionResultFactory(),
+                $runStaticAnalysis,
             );
         };
 
         $reflectionOffsetSet->setAccessible(true);
-        $reflectionOffsetSet->invokeArgs($container, [MutantExecutionResultFactory::class, $factory]);
+        $reflectionOffsetSet->invokeArgs($new_container, [MutantExecutionResultFactory::class, $factory]);
 
-        return $container;
+        return $new_container;
```

Comment:
I don't like this, but it allows for lazy initialization 🤷

Reply:
I don't fully understand what is going on 🤔

Reply:
`getMutantExecutionResultFactory` ( or any other getter ) reads config, config is initialized once.

If we call it too early ( as we did before ), configuration will be initialized using default values, instead of reading from CLI arguments/flags.

so here, we make sure that `getMutantExecutionResultFactory` is not called early to avoid that.

however, if we used the same instance, we will have an infinite loop, since the factory will keep calling it self, while we want the old factory to be executed, and that's what is happening here, `$container` still holds a reference to the old factory, hence calling `getMutantExecutionResultFactory` on it results in the old `MutantExecutionResultFactory`.

so yea, it's a bit ugly and hard to read, but works.